kotlin/workflow-ui-android/src/main/java/com/squareup/workflow/ui/WorkflowActivityRunner.kt

```diff
@@ -0,0 +1,146 @@
+import com.squareup.workflow.Workflow
+import io.reactivex.Observable
+import android.arch.lifecycle.ViewModel as AndroidViewModel
+
+/**
+ * Packages a [Workflow] and a [ViewRegistry] to drive an [Activity][FragmentActivity].
+ *
+ * You'll never instantiate one of these yourself. Instead, use
+ * [FragmentActivity.setContentWorkflow]. See that method for more details.
+ */
+class WorkflowActivityRunner<OutputT : Any, RenderingT : Any>
+internal constructor(private val model: WorkflowViewModel<OutputT, RenderingT>) {
+
+  internal val renderings: Observable<out RenderingT> = model.updates.map { it.rendering }
+
+  val viewRegistry: ViewRegistry = model.viewRegistry
+
+  /**
+   * A stream of the [output][OutputT] values emitted by the [Workflow]
+   * managed by this model.
+   */
+  val output: Observable<out OutputT> = model.updates.filter { it.output != null }
+      .map { it.output!! }
+
+  /**
+   * Returns a [Parcelable] instance of [PickledWorkflow] to be written
+   * to the bundle passed to [FragmentActivity.onSaveInstanceState].
+   * Read it back out in [FragmentActivity.onCreate], to serve as the
+   * final argument to [FragmentActivity.setContentWorkflow].
+   */
+  val asParcelable: Parcelable get() = PickledWorkflow(model.lastSnapshot)
+
+  /**
+   * To be called from [FragmentActivity.onBackPressed], to give the managed
+   * [Workflow] access to back button events.
+   *
+   * e.g.:
+   *
+   *    override fun onBackPressed() {
+   *      if (!workflowViewModel.onBackPressed(this)) super.onBackPressed()
+   *    }
+   */
+  fun onBackPressed(activity: Activity): Boolean {
+    return HandlesBack.Helper.onBackPressed(activity.findViewById(R.id.workflow_layout))
+  }
+}
+
+/**
+ * Call this method from [FragmentActivity.onCreate], instead of [FragmentActivity.setContentView].
+ * It creates a [WorkflowActivityRunner] for this activity, if one doesn't already exist, and
+ * sets a view driven by that model as the content view.
+ *
+ * Hold onto the [WorkflowActivityRunner] returned and:
+ *
+ *  - Call [WorkflowActivityRunner.onBackPressed] from [FragmentActivity.onBackPressed] to allow
+ *    workflows to handle back button events. (See [HandlesBack] for more details.)
+ *
+ *  - Write [WorkflowActivityRunner.asParcelable] to the bundle passed to
+ *    [FragmentActivity.onSaveInstanceState]. You'll read that [parcelable][PickledWorkflow]
+ *    back in [FragmentActivity.onCreate], for use by the next call to this method.
+ *
+ *  e.g.:
+ *
+ *     class MainActivity : AppCompatActivity() {
+ *       private lateinit var runner: WorkflowRunner<*, *>
+ *
+ *       override fun onCreate(savedInstanceState: Bundle?) {
+ *         super.onCreate(savedInstanceState)
+ *
+ *         val restored = savedInstanceState?.getParcelable<PickledWorkflow>("WORKFLOW")
+ *         runner = setContentWorkflow(MyViewRegistry, MyRootWorkflow(), restored)
+ *       }
+ *
+ *       override fun onBackPressed() {
```

Comment:
Note that there is also a new callback for this: https://developer.android.com/reference/androidx/activity/OnBackPressedCallback

Reply:
Whoa, that's huge. We can get rid of HandlesBack entirely. Requires AndroidX, right?

Reply:
#250

Reply:
Yes, I'm pretty sure it requires AndroidX.